test/Main.hs

```diff
@@ -43,72 +56,80 @@ data OpaqueType = OpaqueType Int
+    [ testGroup
+        "Instances Tests"
+        [ testCase "Display instance for Text stays the same" $
+            display ("3" :: Text) @?= ("3" :: Text)
+        , testCase "Deriving via its own Show instance works" $
+            T.unpack (display AD) @?= show AD
+        , testCase "Opaque types stay opaque" $
+            display (OpaqueType 3 :: OpaqueType) @?= "<opaque>"
+        , testCase "Manual instance is stays the same" $
+            display (MT 32) @?= "MT 32"
+        , testCase "List instance is equivalent to Show" $ do
+            let list = [1 .. 5] :: [Int]
+            T.unpack (display list) @?= show list
+        , testCase "Single-element List instance is equivalent to Show" $ do
+            let list = [1] :: [Int]
+            T.unpack (display list) @?= show list
+        , testCase "List instance is streamed lazily" $ do
+            let list = [1 ..] :: [Int]
+            TL.take 20 (TB.toLazyText $ displayBuilder list) `shouldEvaluateWithin` 100000
+        , testCase "NonEmpty instance is equivalent to Show" $ do
+            let ne = NE.fromList [1 .. 5] :: NonEmpty Int
+            T.unpack (display ne) @?= show ne
+        , testCase "Just True instance is equivalent to Show" $ do
+            T.unpack (display (Just True)) @?= show (Just True)
+        , testCase "Nested Maybe instance is equivalent to Show" $ do
+            let nestedMaybe = Just (Just 5) :: Maybe (Maybe Int)
+            T.unpack (display nestedMaybe) @?= show nestedMaybe
+        , testCase "Nothing instance is equivalent to Show" $ do
+            T.unpack (display (Nothing @Bool)) @?= show (Nothing @Bool)
+        , testCase "Char '\'' instance is equivalent to Text" $ do
+            display '\'' @?= T.singleton '\''
+        , testCase "2-Tuple instance is equivalent to Show" $ do
+            let tuple = (1 :: Int, True)
+            T.unpack (display tuple) @?= show tuple
+        , testCase "3-Tuple instance is equivalent to Show" $ do
+            let tuple = (1 :: Int, True, "hahahha" :: String)
+            display tuple @?= "(" <> display (1 :: Int) <> "," <> display True <> "," <> display @String "hahahha" <> ")"
+        ]
+    , testGroup
+        "Properties Tests"
+        [ testProperty "Text instance stays the same" $ do
+            \string -> display (string :: Text) === string
+        , testProperty "String instance is equivalent to Text" $ do
+            \string -> display (string :: String) === T.pack string
+        , testProperty "Chars are packed" $
+            \c -> display (c :: Char) === T.singleton c
+        ]
+    , testGroup
+        "Forbidden Instances"
+        [ testCase "Should not compile for a function instance" $
+            shouldNotTypecheck (display id) -- `shouldThrow` SomeException
+        , testCase "Should not compile for ByteStrings" $
+            let bs = "badstring" :: ByteString
+             in shouldNotTypecheck (display bs) -- `shouldThrow` SomeException
+        ]
```

Comment:
@MorrowM I'm really confused as to why this test group suddenly fails. Would you mind enlightening me on this?

Reply:
I have done some research and testing. Turns out this test group never actually worked. 
`-fdefer-type-errors` will defer type errors for as long as possible. This includes ignoring unused lifted constraints. The `TypeError` constraints are lifted and never used ergo they're never triggered.

So why are the tests succeeding up until now? Well, we're catching the`errorCall`s from the `undefined` stubs in the method implementations. `shouldNotTypecheck` isn't actually doing anything of substance. Change it to the following (or remove the and the tests fail:
```hs
data CustomErr = CustomErr
  deriving (Eq, Show)
instance Exception CustomErr

instance CannotDisplayBareFunctions => Display (a -> b) where
  displayBuilder = throw CustomErr
```
I suppose just catching the `ErrorCall` (or perhaps a custom exception type) works since there would need to be a catastrophic GHC bug in order for someone to be able to call the method without deferring type errors.

Relevant GHC issue: https://gitlab.haskell.org/ghc/ghc/-/issues/16249

Reply:
Thanks a lot @MorrowM! I think I will remove those tests for now, and restore them when we have a more reliable way to move forward.